### PR TITLE
add err handling  in /test/command/

### PR DESCRIPTION
**Pull request by tanjunchen**
### Ⅰ. Describe what this PR did
add error handling

### Ⅱ. Does this pull request fix one issue?

Ref: #952
### Ⅳ. Describe how to verify it
run ```make integration-test```
result  ``` ok  	github.com/dragonflyoss/Dragonfly/test  ```
run ```./hack/unit-test.sh```
```TEST: unit test
?   	github.com/dragonflyoss/Dragonfly/apis/types	[no test files]
ok  	github.com/dragonflyoss/Dragonfly/client	1.500s	coverage: 34.3% of statements
?   	github.com/dragonflyoss/Dragonfly/cmd/dfdaemon	[no test files]
ok  	github.com/dragonflyoss/Dragonfly/cmd/dfdaemon/app	1.424s	coverage: 41.1% of statements
?   	github.com/dragonflyoss/Dragonfly/cmd/dfget	[no test files]
ok  	github.com/dragonflyoss/Dragonfly/cmd/dfget/app	1.421s	coverage: 54.3% of statements
?   	github.com/dragonflyoss/Dragonfly/cmd/supernode	[no test files]
ok  	github.com/dragonflyoss/Dragonfly/cmd/supernode/app	1.430s	coverage: 52.0% of statements
?   	github.com/dragonflyoss/Dragonfly/dfdaemon	[no test files]
ok  	github.com/dragonflyoss/Dragonfly/dfdaemon/config	1.033s	coverage: 100.0% of statements
?   	github.com/dragonflyoss/Dragonfly/dfdaemon/constant	[no test files]
?   	github.com/dragonflyoss/Dragonfly/dfdaemon/downloader	[no test files]
?   	github.com/dragonflyoss/Dragonfly/dfdaemon/downloader/dfget	[no test files]
?   	github.com/dragonflyoss/Dragonfly/dfdaemon/exception	[no test files]
?   	github.com/dragonflyoss/Dragonfly/dfdaemon/handler	[no test files]
ok  	github.com/dragonflyoss/Dragonfly/dfdaemon/proxy	1.018s	coverage: 9.1% of statements
?   	github.com/dragonflyoss/Dragonfly/dfdaemon/transport	[no test files]
ok  	github.com/dragonflyoss/Dragonfly/dfget/config	1.033s	coverage: 91.5% of statements
ok  	github.com/dragonflyoss/Dragonfly/dfget/core	1.409s	coverage: 53.2% of statements
ok  	github.com/dragonflyoss/Dragonfly/dfget/core/api	1.413s	coverage: 46.3% of statements
ok  	github.com/dragonflyoss/Dragonfly/dfget/core/downloader	1.682s	coverage: 100.0% of statements
ok  	github.com/dragonflyoss/Dragonfly/dfget/core/downloader/back_downloader	1.447s	coverage: 78.4% of statements
ok  	github.com/dragonflyoss/Dragonfly/dfget/core/downloader/p2p_downloader	3.429s	coverage: 12.7% of statements
ok  	github.com/dragonflyoss/Dragonfly/dfget/core/helper	1.417s	coverage: 4.2% of statements
ok  	github.com/dragonflyoss/Dragonfly/dfget/core/regist	8.909s	coverage: 89.7% of statements
ok  	github.com/dragonflyoss/Dragonfly/dfget/core/uploader	5.978s	coverage: 90.9% of statements
ok  	github.com/dragonflyoss/Dragonfly/dfget/types	1.015s	coverage: 73.5% of statements
ok  	github.com/dragonflyoss/Dragonfly/dfget/util	1.021s	coverage: 69.6% of statements
ok  	github.com/dragonflyoss/Dragonfly/hack/boilerplate	1.012s	coverage: 44.7% of statements
?   	github.com/dragonflyoss/Dragonfly/pkg/atomiccount	[no test files]
?   	github.com/dragonflyoss/Dragonfly/pkg/constants	[no test files]
ok  	github.com/dragonflyoss/Dragonfly/pkg/dflog	1.025s	coverage: 56.5% of statements
?   	github.com/dragonflyoss/Dragonfly/pkg/digest	[no test files]
ok  	github.com/dragonflyoss/Dragonfly/pkg/errortypes	1.011s	coverage: 14.3% of statements
ok  	github.com/dragonflyoss/Dragonfly/pkg/fileutils	1.054s	coverage: 86.3% of statements
ok  	github.com/dragonflyoss/Dragonfly/pkg/httputils	1.338s	coverage: 63.9% of statements
?   	github.com/dragonflyoss/Dragonfly/pkg/limitreader	[no test files]
?   	github.com/dragonflyoss/Dragonfly/pkg/metricsutils	[no test files]
ok  	github.com/dragonflyoss/Dragonfly/pkg/netutils	1.032s	coverage: 56.1% of statements
?   	github.com/dragonflyoss/Dragonfly/pkg/printer	[no test files]
ok  	github.com/dragonflyoss/Dragonfly/pkg/queue	1.300s	coverage: 100.0% of statements
ok  	github.com/dragonflyoss/Dragonfly/pkg/rate	1.014s	coverage: 80.9% of statements
ok  	github.com/dragonflyoss/Dragonfly/pkg/ratelimiter	8.518s	coverage: 91.4% of statements
?   	github.com/dragonflyoss/Dragonfly/pkg/stat	[no test files]
ok  	github.com/dragonflyoss/Dragonfly/pkg/stringutils	1.012s	coverage: 100.0% of statements
ok  	github.com/dragonflyoss/Dragonfly/pkg/syncmap	1.011s	coverage: 24.1% of statements
?   	github.com/dragonflyoss/Dragonfly/pkg/timeutils	[no test files]
ok  	github.com/dragonflyoss/Dragonfly/pkg/util	1.014s	coverage: 90.9% of statements
ok  	github.com/dragonflyoss/Dragonfly/supernode/config	1.021s	coverage: 46.2% of statements
?   	github.com/dragonflyoss/Dragonfly/supernode/daemon	[no test files]
?   	github.com/dragonflyoss/Dragonfly/supernode/daemon/mgr	[no test files]
ok  	github.com/dragonflyoss/Dragonfly/supernode/daemon/mgr/cdn	1.439s	coverage: 30.1% of statements
ok  	github.com/dragonflyoss/Dragonfly/supernode/daemon/mgr/dfgettask	1.412s	coverage: 48.9% of statements
?   	github.com/dragonflyoss/Dragonfly/supernode/daemon/mgr/gc	[no test files]
?   	github.com/dragonflyoss/Dragonfly/supernode/daemon/mgr/mock	[no test files]
ok  	github.com/dragonflyoss/Dragonfly/supernode/daemon/mgr/peer	1.398s	coverage: 84.6% of statements
?   	github.com/dragonflyoss/Dragonfly/supernode/daemon/mgr/pieceerror	[no test files]
ok  	github.com/dragonflyoss/Dragonfly/supernode/daemon/mgr/progress	1.408s	coverage: 18.5% of statements
ok  	github.com/dragonflyoss/Dragonfly/supernode/daemon/mgr/scheduler	1.402s	coverage: 20.8% of statements
ok  	github.com/dragonflyoss/Dragonfly/supernode/daemon/mgr/task	1.425s	coverage: 23.6% of statements
ok  	github.com/dragonflyoss/Dragonfly/supernode/daemon/util	1.015s	coverage: 85.7% of statements
?   	github.com/dragonflyoss/Dragonfly/supernode/httpclient	[no test files]
?   	github.com/dragonflyoss/Dragonfly/supernode/httpclient/mock	[no test files]
ok  	github.com/dragonflyoss/Dragonfly/supernode/plugins	1.021s	coverage: 94.1% of statements
ok  	github.com/dragonflyoss/Dragonfly/supernode/server	2.637s	coverage: 19.8% of statements
ok  	github.com/dragonflyoss/Dragonfly/supernode/store	1.027s	coverage: 68.6% of statements
ok  	github.com/dragonflyoss/Dragonfly/supernode/util	1.012s	coverage: 47.8% of statements
?   	github.com/dragonflyoss/Dragonfly/version	[no test files]
```

### CODE REVIEW

**Comment by tanjunchen** [comment]
@Starnop Can you help me review this ?

**Comment by yeya24** [comment]
> @Starnop Can you help me review this ? the unit-test runs ok in local runtime.

Please make sure you can pass this integration-test. Run `make integration-test`

File: test/command/cmd.go [hunk 1]
```diff
@@ -60,6 +61,7 @@ func init() {
 	supernodePath = fp.Join(binDir, "supernode")
 }
 
+// checkExist checks the path s whether is existing.
```
Comment (Starnop) [inline test/command/cmd.go]: The `s` is weird. Could you please update the param name to path or something else? THX.

Comment (tanjunchen) [reply test/command/cmd.go]: OK  My mistake

**Comment by tanjunchen** [comment]
> > @Starnop Can you help me review this ? the unit-test runs ok in local runtime.
> 
> Please make sure you can pass this integration-test. Run `make integration-test`

OK

**Comment by tanjunchen** [comment]
@yeya24  
image

**Comment by fengzixu** [comment]
> @yeya24
> 
> image
> 
> image

Please move the test result into the "Describe how to verify it" section in the description of PR.

**Comment by tanjunchen** [comment]
@fengzixu OK

File: test/command/cmd.go [hunk 2]
```diff
@@ -213,7 +215,9 @@ func (s *Starter) KillAll() {
 func (s *Starter) Clean() {
 	s.KillAll()
 	if s.Home != "" {
-		os.RemoveAll(s.Home)
+		if err := os.RemoveAll(s.Home); err != nil {
```
Comment (allencloud) [inline test/command/cmd.go]: I do not think every error should be declared and printed. Even printed, not necessary as a Error level.
You can see lots of docs, like:
https://travix.io/unhandled-errors-in-go-3f341f2704dd

Comment (tanjunchen) [reply test/command/cmd.go]: Sure.I will think about that more.what is your  advice? ignore or  lower level?

Comment (tanjunchen) [reply test/command/cmd.go]: thanks your sharings

File: test/command/cmd.go [hunk 3]
```diff
@@ -213,7 +214,9 @@ func (s *Starter) KillAll() {
...
+			fmt.Printf("failed to clean all temporary directories and processes: %s", s.Home)
```
Comment (tanjunchen) [inline test/command/cmd.go]: another view : panic

File: test/command/cmd.go [hunk 4]
```diff
@@ -223,14 +226,18 @@ func (s *Starter) kill(cmd *exec.Cmd) {
 		return
 	}
 	if cmd.ProcessState == nil {
-		cmd.Process.Signal(os.Interrupt)
+		if err := cmd.Process.Signal(os.Interrupt); err != nil {
+			fmt.Printf("failed to send a signal to the Process: %v", err)
```
Comment (tanjunchen) [inline test/command/cmd.go]: ditto

File: test/command/cmd.go [hunk 5]
```diff
@@ -223,14 +226,18 @@ func (s *Starter) kill(cmd *exec.Cmd) {
...
+		}
 	}
 	if v, ok := s.listMap[cmd]; ok {
 		s.cmdList.Remove(v)
 		delete(s.listMap, cmd)
 	}
 	if v, ok := s.fileSrv[cmd]; ok {
-		v.Shutdown(context.Background())
+		if err := v.Shutdown(context.Background()); err != nil {
```
Comment (tanjunchen) [inline test/command/cmd.go]: ditto

**Pull request closed without merging**